sentry-android-core/src/main/java/io/sentry/android/core/internal/util/ScreenshotUtils.java

```diff
@@ -132,7 +157,19 @@ public class ScreenshotUtils {
           return null;
         }
       }
+      return bitmap;
+    } catch (Throwable e) {
+      logger.log(SentryLevel.ERROR, "Taking screenshot failed.", e);
+    }
+    return null;
+  }
 
+  public static @Nullable byte[] compressBitmapToPng(
+      final @Nullable Bitmap bitmap, final @NotNull ILogger logger) {
+    if (bitmap == null || bitmap.isRecycled()) {
+      return null;
+    }
+    try (final ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream()) {
       // 0 meaning compress for small size, 100 meaning compress for max quality.
       // Some formats, like PNG which is lossless, will ignore the quality setting.
       bitmap.compress(Bitmap.CompressFormat.PNG, 0, byteArrayOutputStream);
```

Comment:
It's worth mentioning that we never call `bitmap.recycle()` - and we also didn't do before. But I guess it makes sense to do so!

Reply:
yeah, since we also create a new bitmap for every new screenshot we should probably recycle 😅